### PR TITLE
Reorg corelib code

### PR DESCRIPTION
**Pull request by chenhao94**
Cris-core consists of multiple relatively-independent components and should not be considered a monolithic library. Especially, when we add more stuff to it, the users may only need to depend on the part of cris-core. So we may want to split it into multiple libs.

### CODE REVIEW

File: utils/time.cc [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#include "cris/core/timer.h"
+#include "cris-core/time.h"
```
Comment (xkszltl) [inline utils/time.cc:1]: Why changing cris/core to cris-core? cris/core seems fine to me, and it's cleaner.
And regardless of the reason, you'll need to update clang-format rules for enforce the ordering.

File: utils/time.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace cris::core {
+
+using cr_timestamp_nsec_t = std::int64_t;
+using cr_duration_nsec_t  = std::int64_t;
+
+// CPU Timestamp counter tick
+unsigned long long GetTSCTick(unsigned& aux);
+
+// monotonic timestamp
```
Comment (xkszltl) [inline utils/time.h:13]: Cap and .

File: utils/time.h [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <cstdint>
+
+namespace cris::core {
+
+using cr_timestamp_nsec_t = std::int64_t;
+using cr_duration_nsec_t  = std::int64_t;
+
+// CPU Timestamp counter tick
```
[the remark this answers is not in the document]
Comment (xkszltl) [reply utils/time.h:10]: And what's a "timestamp counter"...

File: BUILD [hunk 4]
```diff
@@ -3,19 +3,53 @@ package(default_visibility = ["//visibility:public"])
...
+        "@com_github_google_glog//:glog",
+    ],
+)
+
+cris_cc_library (
+    name = "core",
+    srcs = glob(["core/**/*.cc"]),
+    hdrs = glob(["core/**/*.h"]),
+    include_prefix = "cris-core",
+    strip_include_prefix = "core",
```
[the remark this answers is not in the document]
Comment (chenhao94) [reply BUILD:21]: I also hate this, so let's remove it.

The only pros of renaming is we can shorten the include lines while keeping the files structured.

File: utils/time.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (xkszltl) [reply utils/time.h]: Probably you can cap in a better way....

**Pull request merged**